core/camel-health/src/main/java/org/apache/camel/impl/health/HealthCheckRegistryRepository.java

```diff
@@ -109,14 +109,10 @@ private HealthCheck toHealthCheck(HealthCheck hc) {
     }
 
     private HealthCheckConfiguration matchConfiguration(String id) {
-        if (configurations != null) {
-            for (String key : configurations.keySet()) {
-                if (PatternHelper.matchPattern(id, key)) {
-                    return configurations.get(key);
-                }
-            }
-        }
-        return fallbackConfiguration;
-    }
 
+        return configurations.values().stream()
+                .filter(s -> PatternHelper.matchPattern(id, s.getParent()))
+                .findAny()
+                .orElseGet(() -> fallbackConfiguration);
```

Comment:
I think `orElseGet(Supplier)` can be replaced by `orElse(fallbackConfiguration)` since the `fallbackConfiguration` does not need to be computed